src/Place/ReadModel/Duplicate/LookupDuplicatePlaceWithSapi3.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Place\ReadModel\Duplicate;
+
+use CultuurNet\UDB3\Model\Place\Place;
+use CultuurNet\UDB3\ReadModel\DocumentDoesNotExist;
+use CultuurNet\UDB3\ReadModel\DocumentRepository;
+use CultuurNet\UDB3\Search\Sapi3SearchService;
+
+class LookupDuplicatePlaceWithSapi3 implements LookupDuplicatePlace
+{
+    private Sapi3SearchService $sapi3SearchService;
+    private UniqueAddressIdentifierFactory $addressIdentifierFactory;
+    private DocumentRepository $placeRepository;
+    private string $currentUserId;
+
+    public function __construct(
+        Sapi3SearchService $sapi3SearchService,
+        UniqueAddressIdentifierFactory $addressIdentifierFactory,
+        DocumentRepository $placeRepository,
+        string $currentUserId
+    ) {
+        $this->sapi3SearchService = $sapi3SearchService;
+        $this->addressIdentifierFactory = $addressIdentifierFactory;
+        $this->placeRepository = $placeRepository;
+        $this->currentUserId = $currentUserId;
+    }
+
+    /*
+    When there is only one place propose that place
+    When there are multiple places propose the one with duplicatedBy
+    When there are multiple but no canonical returns the search query to get all places
+    */
+    public function getDuplicatePlaceUri(Place $place): ?string
+    {
+        $results = $this->sapi3SearchService->search(
+            'unique_address_identifier:' . $this->addressIdentifierFactory->hash(
+                $place->getTitle()->getTranslation($place->getMainLanguage())->toString(),
+                $place->getAddress()->getTranslation($place->getMainLanguage()),
+                $this->currentUserId
+            ),
+            1
+        );
+
+        if ($results->getTotalItems() === 0) {
+            return null;
+        }
+
+        if ($results->getTotalItems() === 1) {
+            return $results->getItems()[0]->getUrl()->toString();
+        }
+
+        foreach ($results->getItems() as $item) {
+            try {
+                $placeAsJson = $this->placeRepository->fetch($item->getId())->getAssocBody();
+
+                if (!empty($placeAsJson['duplicatedBy'])) {
+                    return $placeAsJson['duplicatedBy'];
+                }
+            } catch (DocumentDoesNotExist $e) {
+            }
+        }
+
+        if ($this->sapi3SearchService->getLastRequestedUri() === null) {
+            // This should never happen, but probably not worth it to crash the Place API request for it
+            return $results->getItems()[0]->getUrl()->toString();
+        }
+
+        return $this->sapi3SearchService->getLastRequestedUri()->__toString();
```

Comment:
I would never return the string of the SAPI3 query here. The responsibility for `getDuplicatePlaceUri` is just to return the duplicate place. Suppose there are more than 1 duplicate places throw an error.

In the calling code catch the error and then display the query to the API consumer. That approach also does not require a change to `sapi3SearchService` because the caller knows the query.

Reply:
I am not sure I understand.
Instead of return a string here you want me to throw an exception?
But then how does the exception handler know the query?
That information is encapsulated in sapi3SearchService.

Or do you want me to alter sapi3SearchService to throw an error when more then 1 result is found?
I would have to be careful then, that classes is used in a lot of places.

Reply:
On line https://github.com/cultuurnet/udb3-backend/pull/1556/files/08862fb23a6cf9b50de2c0e488b190b6e1aa5fc5#diff-979f1a5d2f82f93de80342e71bc787a2748dda27c1bf5d9372fa50c658f4fe2aR38 you create the query so you can use that value to throw a new exception, something like `CouldNotDetermineOriginalPlace`